readthedocs/proxito/views/utils.py

```diff
@@ -60,7 +60,8 @@ def _get_project_data_from_request(
         )
 
     # Handle single version by grabbing the default version
-    if final_project.single_version:
+    # We might have version_slug when we're serving a PR
+    if final_project.single_version and not version_slug:
```

Comment:
Good catch!

Reply:
I just realize that we should add `self.version_type == EXTERNAL` here. It will be more explicit in the code (not just the comment) but also safer. cc @ericholscher